### PR TITLE
Adds specified give control option to mob VV

### PR DESCRIPTION
**Pull request by kriskog**
## About The Pull Request
Saw some admins with bad habits, this should sort them out. Basically Assume Direct Control except you get to pick the new controller from a list.

## Why It's Good For The Game
Less direct ckey editing

## Changelog
:cl: Skoglol
admin: New mob VV option: Give direct control. Lets you pick a player to put in control of the mob.
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
Nice, but could this be triggered by clicking on the ckey field of the vv panel, the same way clicking the name field triggers a special rename thing?

**Comment by kriskog** [comment]
You guys have some weird fucking habits. :clown_face: 
Ill look into it.

**Comment by kriskog** [comment]
There you go, you can now click the mobs ckey to run this proc.

File: code/modules/admin/verbs/debug.dm [hunk 1]
```diff
@@ -244,6 +244,24 @@ But you can call procs that are of type /mob/living/carbon/human/proc/ for that
...
+	if(M.ckey)
+		if(alert("This mob is being controlled by [M.key]. Are you sure you wish to give away control of it? [M.key] will be made a ghost.",,"Yes","No") != "Yes")
+			return
+		else
+			M.ghostize(FALSE)
+	var/client/newkey = input(src, "Pick the player to put in control.", "New player") as null|anything in sortList(GLOB.clients)
+	var/mob/oldmob = newkey.mob
+	M.ckey = newkey.key
+	if(isobserver(oldmob) || alert("Do you want to delete [newkey]'s old mob?","Delete?","Yes","No") != "No")
+		qdel(oldmob)
```
Comment (Cyberboss) [inline code/modules/admin/verbs/debug.dm]: No actions should be taken until all input prompts are dealt with.

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/admin/verbs/debug.dm (hunk 1 above)
Comment (kriskog) [reply code/modules/admin/verbs/debug.dm]: Good call. Added some more checks after the inputs/alerts both in this proc and the old one.

File: code/modules/admin/verbs/debug.dm [hunk 2]
```diff
@@ -234,16 +234,44 @@ But you can call procs that are of type /mob/living/carbon/human/proc/ for that
...
-			M.ghostize(FALSE)
+	if(!M || QDELETED(M))
+		to_chat(usr, "<span class='warning'>The target mob no longer exists.</span>")
+		return
+	var/mob/adminmob = mob
+	M.ghostize(FALSE)
+	M.ckey = ckey
+	if(isobserver(adminmob))
+		qdel(adminmob)
 	message_admins("<span class='adminnotice'>[key_name_admin(usr)] assumed direct control of [M].</span>")
```
Comment (AnturK) [inline code/modules/admin/verbs/debug.dm]: This is post ckey transfer, usr is the keyless mob now.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/admin/verbs/debug.dm (hunk 2 above)
Comment (kriskog) [reply code/modules/admin/verbs/debug.dm]: Thanks

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -1373,7 +1373,7 @@
 	. = ..()
 	var/refid = REF(src)
 	. += {"
-		<br><font size='1'>[VV_HREF_TARGETREF_1V(refid, VV_HK_BASIC_EDIT, "[ckey || "no ckey"]", NAMEOF(src, ckey))] / [VV_HREF_TARGETREF_1V(refid, VV_HK_BASIC_EDIT, "[real_name || "no real name"]", NAMEOF(src, real_name))]</font>
+		<br><font size='1'>[VV_HREF_TARGETREF_1V(refid, VV_HK_GIVE_DIRECT_CONTROL, "[ckey || "no ckey"]", NAMEOF(src, ckey))] / [VV_HREF_TARGETREF_1V(refid, VV_HK_BASIC_EDIT, "[real_name || "no real name"]", NAMEOF(src, real_name))]</font>
```
Comment (AnturK) [inline code/modules/mob/living/living.dm]: Pretty sure this doesn't need three param define now, just two ?

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/living.dm (hunk 3 above)
Comment (kriskog) [reply code/modules/mob/living/living.dm]: Yup

**Pull request merged**